frontend/packages/kubevirt-plugin/src/plugin.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'lodash-es';
+import { Plugin, ResourceNSNavItem, ResourceListPage, ResourceDetailPage, ModelFeatureFlag, YamlTemplate, ModelDefinition } from '@console/plugin-sdk';
+
+import * as models from './models';
+import { vmYamlTemplate } from './yaml-templates';
+
+type ConsumedExtensions = ResourceNSNavItem | ResourceListPage | ResourceDetailPage | ModelFeatureFlag | YamlTemplate | ModelDefinition;
+
+const FLAG_KUBEVIRT = 'KUBEVIRT';
+
+const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'ModelDefinition',
+    properties: {
+      models: _.values(models),
+    },
+  },
+  {
+    type: 'FeatureFlag/Model',
+    properties: {
+      model: models.VirtualMachineModel,
+      flag: FLAG_KUBEVIRT,
+    },
+  },
+  {
+    type: 'NavItem/ResourceNS',
+    properties: {
+      section: 'Workloads',
+      componentProps: {
+        name: 'Virtual Machines',
+        resource: models.VirtualMachineModel.plural,
+        required: FLAG_KUBEVIRT,
+      },
+      mergeAfter: 'Pods', // rendered name of the resource navigation link in the left-side menu
```

Comment:
Nav extension properties will be documented via #1638 so we can remove this comment, I think :smiley:

Reply:
done